gpu/segmentation/include/pcl/gpu/segmentation/impl/gpu_extract_clusters.hpp

```diff
@@ -96,71 +124,57 @@ pcl::gpu::extractEuclideanClusters (const pcl::PointCloud<pcl::PointXYZ>::Ptr  &
       {
-        std::cout << " CPU: ";
+        PCL_DEBUG(" CPU: ");
+        //data.clear();
         for(std::size_t p = 0; p < queries_host.size (); p++)
         {
           // Execute the radiusSearch on the host
-          tree->radiusSearchHost(queries_host[p], tolerance, data, max_answers);
-        }
-        // Store the previously found number of points
-        previous_found_points = found_points;
-        // Clear queries list
-        queries_host.clear();
-
-        //std::unique(data.begin(), data.end());
-        if(data.size () == 1)
-          continue;
-
-        // Process the results
-        for(std::size_t i = 0; i < data.size (); i++)
-        {
-          if(processed[data[i]])
-            continue;
-          processed[data[i]] = true;
-          queries_host.push_back ((*host_cloud_)[data[i]]);
-          found_points++;
-          r.indices.push_back(data[i]);
+          cpu_tmp.clear();
+          tree->radiusSearchHost(queries_host[p], tolerance, cpu_tmp, max_answers);
+          std::copy(cpu_tmp.begin(), cpu_tmp.end(), std::back_inserter(data));
         }
       }
-
       // If number of queries is high enough do it here
       else
       {
-        std::cout << " GPU: ";
+        PCL_DEBUG(" GPU: ");
+        sizes.clear();
         // Copy buffer
         queries_device = DeviceArray<PointXYZ>(queries_device_buffer.ptr(),queries_host.size());
         // Move queries to GPU
         queries_device.upload(queries_host);
         // Execute search
         tree->radiusSearch(queries_device, tolerance, max_answers, result_device);
         // Copy results from GPU to Host
-        result_device.sizes.download (sizes);
-        result_device.data.download (data);
-        // Store the previously found number of points
-        previous_found_points = found_points;
-        // Clear queries list
-        queries_host.clear();
-        for(std::size_t qp = 0; qp < sizes.size (); qp++)
+        result_device.sizes.download(sizes);
+        pcl::detail::economical_download(result_device, sizes, max_answers, data);
+      }
+      // Store the previously found number of points
+      previous_found_points = found_points;
+      // Clear queries list
+      queries_host.clear();
+
+      if(data.size () == 1)
+        continue;
+
+      // Process the results
+      for(int idx : data)
         {
-          for(int qp_r = 0; qp_r < sizes[qp]; qp_r++)
-          {
-            if(processed[data[qp_r + qp * max_answers]])
-              continue;
-            processed[data[qp_r + qp * max_answers]] = true;
-            queries_host.push_back ((*host_cloud_)[data[qp_r + qp * max_answers]]);
-            found_points++;
-            r.indices.push_back(data[qp_r + qp * max_answers]);
-          }
-        }
+        if(processed[idx])
+          continue;
+        processed[idx] = true;
+        queries_host.push_back ((*host_cloud_)[idx]);
+        found_points++;
+        r.indices.push_back(idx);
       }
```

Comment:
The braces need formatting

Reply:
Oh, sorry, I tried to fix it! Is it correct now? I have a question though: When I clang-format the file `gpu_extract_clusters.hpp` many lines of the file change. Am I doing something wrong or does the same maybe also happen to you too?

Reply:
`gpu` module isn't formatted yet. So your experience is correct. [The plugin I use](https://github.com/rhysd/vim-clang-format) allows me to format only a selection of lines instead of all the lines in the file.

You can either:
* use similar settings for some plugin in your fav editor
* create an additional formatting commit (at the start or end) so we can skip that/select that easily during code-review

Reply:
Ha - thanks for pointing out this feature in vim clang-format! I always only format the entire document, never a few lines. Shall I maybe make a batch PR clang-formatting the GPU codebase?